### PR TITLE
fix(node): don't unnecessarily ask for env access

### PR DESCRIPTION
**Pull request by lino-levan**
Props to @iuioiua for getting `querySync` landed (https://github.com/denoland/deno/pull/17019).

Closes https://github.com/denoland/deno_std/issues/2446.

### CODE REVIEW

**Comment by lino-levan** [comment]
Canary is quite literally not built yet, this *should* pass when it is finished.

**Comment by iuioiua** [comment]
Quick turnaround! 🚀😳

**Comment by kt3k** [comment]
Looks good to me. Thanks for your contribution!

But now this is blocked by Deno Deploy compatibility as it doesn't have `querySync` yet. Let's wait until it updates permissions API

**Comment by bartlomieju** [comment]
> BTW currently we exceptionally skip checking `env` permission for this particular env variable. See [denoland/deno#15893](https://github.com/denoland/deno/pull/15893)
> 
> Maybe we should remove that exceptional handling in CLI after this landed. (@bartlomieju What do you think about this?)

Sure, sounds good to me!

File: node/internal/util/debuglog.ts [hunk 1]
```diff
@@ -104,15 +104,12 @@ export function debuglog(
 }
 
 let debugEnv;
-try {
+if (Deno.permissions.querySync({ name: "env", variable: "NODE_DEBUG" })) {
```
Comment (kt3k) [inline node/internal/util/debuglog.ts:107]: This needs to be:

```suggestion
if (Deno.permissions.querySync({ name: "env", variable: "NODE_DEBUG" }).state === "granted") {
```

Comment (lino-levan) [reply node/internal/util/debuglog.ts]: Nice catch!

**Comment by iuioiua** [comment]
Can we re-create this PR in the main repo if it's still needed?

**Comment by lino-levan** [comment]
Closing this PR as it is stale.

**Pull request closed without merging**